keanu-project/src/test/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/operators/binary/ConcatenationVertexTest.java

```diff
@@ -437,4 +438,15 @@ public void canConcatenateAutoDiffOfManyMatricesAlongDimensionZero() {
         assertEquals(concatPartialForward.withRespectTo(f), concatPartialReverse.withRespectTo(f));
     }
 
+    @Test
+    public void changesMatchGradient() {
+        DoubleVertex inputA = new UniformVertex(new int[]{2, 2, 2}, -10.0, 10.0);
+        DoubleVertex inputB = new UniformVertex(new int[]{2, 2, 2}, -10.0, 10.0);
+        DoubleVertex inputC = new UniformVertex(new int[]{2, 2, 2}, -10.0, 10.0);
+        DoubleVertex outputVertex = new ConcatenationVertex(0, inputA, inputB, inputC);
+        final double INCREMENT = 10.0;
+        final double DELTA = 1e-10;
+        finiteDifferenceMatchesGradient(ImmutableList.of(inputA, inputB, inputC), outputVertex, INCREMENT, DELTA);
```

Comment:
Minor point, but I think you can inline these final doubles. You're not losing any information because IntelliJ will display the parameter names for you (`INCREMENT_AMOUNT` and `DELTA`).

Reply:
yeah - actually these are actually just hangovers from before I refactored.  I'll put them inline.